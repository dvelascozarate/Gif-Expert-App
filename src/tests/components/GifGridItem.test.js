import { shallow } from 'enzyme'
import React from 'react'
import { GifGridItem } from '../../components/GifGridItem'

describe('Pruebas al componente GitGridItem', () => {

    const title = 'Hulk'
    const url = 'https://media2.giphy.com/media/aS8ypUweGOXMA/giphy-downsized-medium.gif?cid=a8e6cbe4ly1k6i5op9j5yom7q2yvco00nzk45ax0gb3p06tk&rid=giphy-downsized-medium.gif&ct=g'

    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('Debe mostrar correctamente el componente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar el titulo', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('Debe mostrar la imagen', () => {
        const img = wrapper.find('img');
        expect(img.prop('src').trim()).toBe(url);
        expect(img.prop('alt').trim()).toBe(title);
    })

    test('El contenedor tiene que tener la clase animate__backInRight', () => {
        const div = wrapper.find('div');
        expect(div.prop('className').trim()).toContain('animate__backInRight');
    })


})
