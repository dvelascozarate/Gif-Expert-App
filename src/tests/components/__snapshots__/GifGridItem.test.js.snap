// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GitGridItem Debe mostrar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__backInRight"
>
  <img
    alt="Hulk"
    src="https://media2.giphy.com/media/aS8ypUweGOXMA/giphy-downsized-medium.gif?cid=a8e6cbe4ly1k6i5op9j5yom7q2yvco00nzk45ax0gb3p06tk&rid=giphy-downsized-medium.gif&ct=g"
  />
  <p>
    Hulk
  </p>
</div>
`;
